InnerEye/Scripts/download_model_and_run_scoring.py

```diff
@@ -0,0 +1,91 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+
+import argparse
+import os
+import subprocess
+import sys
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+from azureml.core import Model, Run
+
+
+def spawn_and_monitor_subprocess(process: str, args: List[str], env: Dict[str, str]) -> Tuple[int, List[str]]:
+    """
+    Helper function to spawn and monitor subprocesses.
+    :param process: The name or path of the process to spawn.
+    :param args: The args to the process.
+    :param env: The environment variables for the process (default is the environment variables of the parent).
+    :return: Return code after the process has finished, and the list of lines that were written to stdout by the
+    subprocess.
+    """
+    p = subprocess.Popen(
+        [process] + args,
+        shell=False,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        env=env
+    )
+
+    # Read and print all the lines that are printed by the subprocess
+    stdout_lines = [line.decode('UTF-8').strip() for line in p.stdout]  # type: ignore
+    for line in stdout_lines:
+        print(line)
+
+    # return the subprocess error code to the calling job so that it is reported to AzureML
+    return p.wait(), stdout_lines
+
+
+def run() -> None:
+    """
+    Downloads a model from AzureML, and starts the score script (usually score.py) in the root folder of the model.
+    Downloading the model is only supported if the present code is running inside of AzureML. When running outside
+    of AzureML, the model must have been downloaded beforehand into the folder given by the model-folder argument.
+    The script is executed with the current Python interpreter.
+    If the model requires a specific Conda environment to run in, the caller of this script needs to ensure
+    that this has been set up correctly (taking the environment.yml file stored in the model).
+    All arguments that are not recognized by the present code will be passed through to `score.py` unmodified.
+    Example arguments:
+        download_model_and_run_scoring.py --model-id=Foo:1 start.py --foo=1 --bar
```

Comment:
should we call it just run_scoring? since it is not downloading expect if in AML?

Reply:
The normal use case is for that to run inside AzureML. having the model already there is only used for testing.